src/shims/unix/linux/eventfd.rs

```diff
@@ -163,19 +166,21 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         }
 
         let mut is_nonblock = false;
-        // Unload the flag that we support.
+        // Unset the flag that we support.
         // After unloading, flags != 0 means other flags are used.
         if flags & efd_cloexec == efd_cloexec {
+            // cloexec is ignored because Miri does not support exec.
             flags &= !efd_cloexec;
         }
         if flags & efd_nonblock == efd_nonblock {
             flags &= !efd_nonblock;
             is_nonblock = true;
         }
         if flags != 0 {
-            let einval = this.eval_libc("EINVAL");
-            this.set_last_error(einval)?;
-            return Ok(Scalar::from_i32(-1));
+            throw_unsup_format!(
+                "eventfd: Flags other than EFD_CLOEXEC, EFD_NONBLOCK \
+            and EFD_SEMAPHORE is not supported."
```

Comment:
EFD_SEMAPHORE is also not supported...
```suggestion
                "eventfd: encountered unknown unsupported flags {:#x}"
```

Reply:
Oh yea this gave the illusion of we actually support EFD_SEMAPHORE when in fact, they will hit the unsup EFD_SEMAPHORE branch.